cpp/models/lct_secir/model.h

```diff
@@ -35,233 +35,372 @@ namespace mio
+
+        // Function call for next group if applicable.
+        if constexpr (Group + 1 < m_groups) {
+            get_derivatives_impl<Group + 1>(pop, y, t, dydt);
+        }
     }
 
     /**
-     * @brief Cumulates a simulation result with subcompartments to produce a result that divides the population only
-     *   into the infection states defined in InfectionState.
+     * @brief Calculated the derivative of the Susceptible compartment for Group1.
      *
-     * If the model is used for simulation, we will get a result in form of a TimeSeries with infection states divided 
-     * in subcompartments.
-     * The function calculates a TimeSeries without subcompartmens from another TimeSeries with subcompartments. 
-     * This is done by summing up the numbers in the subcompartments.
-     * @param[in] subcompartments_ts Result of a simulation with the model.
-     * @return Result of the simulation divided in infection states without subcompartments. 
-     *  Returns TimeSeries with values -1 if calculation is not possible.
+     * This is done recursively by calculating the interaction terms with each group.
+     * @tparam Group1 The group for which the derivative of the Susceptible compartment should be calculated. 
+     * @tparam Group2 The group that Group1 interacts with. 
+     * @param[in] pop The current state of the population in the geographic unit we are considering.
+     * @param[in] y The current state of the model (or a subpopulation) as a flat array.
+     * @param[in] t The current time.
+     * @param[out] dydt A reference to the calculated output.
      */
-    TimeSeries<ScalarType> calculate_compartments(const TimeSeries<ScalarType>& subcompartments_ts) const
+    template <
+        size_t Group1, size_t Group2 = 0,
+        std::enable_if_t<(Group1 < m_groups) && (Group1 >= 0) && (Group2 < m_groups) && (Group2 >= 0), bool> = true>
+    void interact(Eigen::Ref<const Eigen::VectorXd> pop, Eigen::Ref<const Eigen::VectorXd> y, ScalarType t,
+                  Eigen::Ref<Eigen::VectorXd> dydt) const
     {
-        TimeSeries<ScalarType> compartments_ts((Eigen::Index)InfectionState::Count);
-        if (!(LctState::Count == subcompartments_ts.get_num_elements())) {
-            log_error("Result does not match infectionState of the Model.");
-            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant((Eigen::Index)InfectionState::Count, -1);
-            compartments_ts.add_time_point(-1, wrong_size);
-            return compartments_ts;
+        using LctState2                 = typename std::tuple_element_t<Group2, tupleLctStates>;
+        size_t Si                       = this->populations.template get_first_index_group<Group1>();
+        ScalarType infectedNoSymptoms_2 = 0;
+        ScalarType infectedSymptoms_2   = 0;
+        auto params                     = this->parameters;
+
+        size_t elem2_first_index = this->populations.template get_first_index_group<Group2>();
```

Comment:
Why did you name it elem? I would add group to the name somewhere, maybe elem2_first_index_group?